server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java

```diff
@@ -73,23 +73,30 @@ public void getCheckpointMetadata(
                 listener.onResponse(new CheckpointInfoResponse(checkpoint, Collections.emptyMap(), null));
                 return;
             }
-            assert mdFile != null : "Remote metadata file can't be null if shard is active " + indexShard.state();
-            metadataMap = mdFile.getMetadata()
-                .entrySet()
-                .stream()
-                .collect(
-                    Collectors.toMap(
-                        e -> e.getKey(),
-                        e -> new StoreFileMetadata(
-                            e.getValue().getOriginalFilename(),
-                            e.getValue().getLength(),
-                            Store.digestToString(Long.valueOf(e.getValue().getChecksum())),
-                            version,
-                            null
+
+            if (mdFile == null) {
+                listener.onResponse(new CheckpointInfoResponse(indexShard.getLatestReplicationCheckpoint(), Collections.emptyMap(), null));
```

Comment:
I think this change is so that SRs can become active and sync with remote before a primary has uploaded anything correct?  I don't think we want to make the same change for other replicas that rely on primary initialization to complete first.

Reply:
I see your point let me handle a case for `isSearchOnly`. Something like ` } else if (indexShard.routingEntry().isSearchOnly()) {`.